client/my-sites/plans-features-main/hooks/use-is-custom-domain-allowed-on-free-plan.ts

```diff
@@ -1,14 +1,21 @@
-import config from '@automattic/calypso-config';
 import { getTld } from 'calypso/lib/domains';
+import { useExperiment } from 'calypso/lib/explat';
 
-// TODO
-// it's not really a hook now, but it eventually will contain a request fetching assigned variations from ExPlat endpoints
-const useIsCustomDomainAllowedOnFreePlan = ( domainName?: string ) => {
-	if ( ! domainName ) {
-		return false;
+const useIsCustomDomainAllowedOnFreePlan = ( domainName?: string ): [ boolean, boolean | null ] => {
+	const [ isLoadingAssignment, experimentAssignment ] = useExperiment(
+		'calypso_onboarding_plans_dotblog_on_free_plan_202307'
+	);
+
+	if ( isLoadingAssignment ) {
+		return [ true, null ];
 	}
 
-	return config.isEnabled( 'domains/blog-domain-free-plan' ) && getTld( domainName ) === 'blog';
+	return [
+		isLoadingAssignment,
+		experimentAssignment?.variationName === 'treatment'
+			? domainName != null && getTld( domainName ) === 'blog'
+			: false,
+	];
```

Comment:
If I am reading this correctly, we'll be assigning users to the experiment even if they do not select a .blog domain. Can we consider using the `options.isEligible` argument of `useExperiment` so that only users who select the .blog domain are assigned? This statement would then become:
```js
const [ isLoadingAssignment, experimentAssignment ] = useExperiment( 'exp_name', { isEligible: domainName != null && getTld( domainName ) === 'blog' } );
```

And then, the return can be simplified to

```js
return [ isLoadingAssignment, experimentAssignment?.variationName === 'treatment' ];
```

Reply:
@aneeshd16 Great suggestion. I've implemented it together with @jdc91 's [comment](https://github.com/Automattic/wp-calypso/pull/78825#discussion_r1274088803) about restricting it in the onboarding flow only.